### ISSUE TITLE
site.yaml resets to default when changing domains

### ISSUE THREAD

**Issue by blizam** [getgrav/grav#3736]
Just had a weird one... was working on a dev site hosted by IP+alias (like: https://100.100.100.100/~alias) and when I cut it over to an actual domain the site.yaml file reverted to the original/defaults. I'd only updated title, author, default email, and the meta description... all were reset to default values.

**Comment by pamtbaau** [comment]
Are you using (unknowingly maybe) [Environment Configuration](https://learn.getgrav.org/17/advanced/environment-config#custom-location-for-the-current-environment)?

Grav reads its configs from several different location:
1. system/config
1. user/config
1. user/env/domain/config (user/domain/config has been deprecated)

Settings found in lower numbers locations will be overridden by higher number locations.

When using 'localhost', Admin will store the config being saved into */user/config*. When using a specific domain, eg. https://mydomain.ext, Admin will store the config file inside */user/env/mydomain.ext/config*.

When moving the site over to another server/domain, eg. https://otherdomain.ext, Grav will first search in */user/env/otherdomain.ext/config* and if not found, it will fallback to */user/config* (which will fallback to */system/config*).

Since you're using a different domain, your settings for https://mydomain.ext will not be read.

**Comment by pamtbaau** [comment]
I've mistakenly pointed you at a subsection of [Environment Configuration](https://learn.getgrav.org/17/advanced/environment-config), so you may have mist the important note at the top of the docs:

> Up to Grav 1.6 environments were stored in user/ folder. Grav 1.7 moves environments to user/env/ to make environments easier to maintain. It is highly recommended that you move all the environments into this new location on your existing sites.

If you say the */user/domain.ext* folder was created on a recent Grav 1.7 version, I have no explanation for that... Best to follow the suggestion from the docs.

**Comment by blizam** [comment]
Ok... going back to your very first question... you had "(unknowingly maybe)" -- which led me to believe there was something on my server (a Virtual Host config) that perhaps DirectAdmin set that caused this all to happen... but in further reading the docs I see that there's no way that could be the case... DirectAdmin isn't going to have any sort of logic to set a GRAV_ENVIRONMENT or GRAV_ENVIRONMENTS_PATH variable... 

So how the heck did a /user/www.domain.com/config/ folder get created in the first place? It's got a security.yaml file in it and /plugins/ folder with 2 plugin yaml files in it... my other core yaml files, and the remainder of my plugin yaml files, are all where I expect them under user/config/

Can I just delete it? Should I chalk this up to something weird DirectAdmin did when I changed the domain... I be a little "too-focused" sometime to be sure but I definitely don't remember ever seeing these folders while I was developing the site... this happened after I changed the domain at the server level (I think).
